src/main/scala/org/scalajs/tools/tsimporter/Main.scala

```diff
@@ -47,18 +57,18 @@ object Main {
     new Importer(output)(definitions, outputPackage)
   }
 
-  private def parseDefinitions(reader: Reader[Char]): List[DeclTree] = {
+  private def parseDefinitions(reader: Reader[Char]): Option[List[DeclTree]] = {
     val parser = new TSDefParser
     parser.parseDefinitions(reader) match {
       case parser.Success(rawCode, _) =>
-        rawCode
+        Some(rawCode)
 
       case parser.NoSuccess(msg, next) =>
         Console.err.println(
             "Parse error at %s\n".format(next.pos.toString) +
             msg + "\n" +
             next.pos.longString)
-        sys.exit(2)
+        None
```

Comment:
If this method does not `exit` anymore, then it really shouldn't `println` either.

Instead of returning an `Option[List[DeclTree]]`, it should return an `Either[String, List[DeclTree]]`, where the `String` is the error message currently inside the `println(...)`. It should be the responsibility of the `main` method to do the `Console.err.println` of that string, followed by the `System.exit(2)`.